include/RAJA/util/View.hpp

```diff
@@ -181,11 +186,17 @@ struct MultiView
   template<bool IsConstView = std::is_const<value_type>::value>
   RAJA_INLINE constexpr MultiView(
       typename std::enable_if<IsConstView, NonConstView>::type const& rhs)
-      : layout(rhs.layout),
-        data(rhs.data)
+      : layout(rhs.get_layout()),
+        data(rhs.get_data())
   {}
 
-  RAJA_INLINE void set_data(pointer_type data_ptr) { data = data_ptr; }
+  RAJA_HOST_DEVICE RAJA_INLINE void set_layout(layout_type&& ly) { layout(ly); }
```

Comment:
Why call the layout with this layout? Also does this need to be an rvalue reference? Layout's don't own any resources, right?
```suggestion
  RAJA_HOST_DEVICE RAJA_INLINE void set_layout(layout_type const& ly) { layout = ly; }
```